kinto/plugins/accounts/views.py

```diff
@@ -43,13 +49,21 @@ class Account(resource.Resource):
     schema = AccountSchema
 
     def __init__(self, request, context):
+        settings = request.registry.settings
         # Store if current user is administrator (before accessing get_parent_id())
-        allowed_from_settings = request.registry.settings.get("account_write_principals", [])
+        allowed_from_settings = settings.get("account_write_principals", [])
         context.is_administrator = (
             len(set(aslist(allowed_from_settings)) & set(request.prefixed_principals)) > 0
         )
         # Shortcut to check if current is anonymous (before get_parent_id()).
         context.is_anonymous = Authenticated not in request.effective_principals
+        # Is the "accounts validation" setting set?
+        context.validation = settings.get("account_validation", False)
+        # Account validation requires the user id to be an email.
+        validation_email_regexp = settings.get(
+            "account_validation.email_regexp", "^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\\.[a-zA-Z0-9-.]+$"
```

Comment:
Just a question: what happens in case I want to update my prez-account-validation existing account? Will my userid be rejected?

Reply:
I'll add a test to make sure an account created before enabling the account validation can still authenticate.